tests/robottelo/test_decorators.py

```diff
@@ -204,13 +204,23 @@ def test_project_mode_different_cases(self, settings):
     def test_invalid_project(self, dec_settings):
         """Assert error is thrown when project has invalid value."""
         dec_settings.project = 'sam'
+
+        @decorators.run_only_on('satddfddffdf')
+        def dummy():
+            pass
+
         with self.assertRaises(decorators.ProjectModeError):
-            decorators.run_only_on('satddfddffdf')
+            dummy()
```

Comment:
@oshtaier @sthirugn as you can see now I must call the decorated function in order to raise the exception. This will be the same thing as when the test runner will be calling the decorated test method.